### ISSUE TITLE
Use the new Photo Picker on Android 11+ in Essentials Picker API

### ISSUE THREAD

**Issue by Redth** [dotnet/maui#9615]
### Description

With the latest [AndroidX Activity release 1.6.0](https://developer.android.com/jetpack/androidx/releases/activity#version_160_2), there's a new API for [picking media](https://developer.android.com/training/data-storage/shared/photopicker) (photos/videos) which will automatically use the newer Android 11+ dialog or fall back to the older intent at runtime for older API levels.

### Public API Changes

No public API changes should be necessary.

### Intended Use-Case

Support the latest UX patterns on Android.

**Comment by jamesmontemagno** [comment]
🔥🔥🔥🔥 yes! <3 :)

**Comment by jfversluis** [comment]
Working on a spec for new MediaPicker stuff, I can include this

**Comment by janseris** [comment]
>The rest of what I'm planning will probably be .NET 8 
So selecting multiple photos in MAUI takes 1 year and 3 months?

**Comment by jfversluis** [comment]
> > The rest of what I'm planning will probably be .NET 8
> 
> So selecting multiple photos in MAUI takes 1 year and 3 months to implement?

No, but it will need breaking API changes which we are only allowed to introduce with major .NET versions

**Comment by salvadorjesus** [comment]
Using the Photo Picker [will be compulsory (*) from August 31th, 2024](https://support.google.com/googleplay/android-developer/answer/13986130), for apps that do not declare the READ_MEDIA_IMAGES and/or the READ_MEDIA_VIDEO permissions.

Declaring those permission **will be forbidden** for "Apps with a one-time or limited use of photo and video files".
[More details about this on Google Play's Photo and Video Permissions policy](https://support.google.com/googleplay/android-developer/answer/14115180). A common example is an app that allows you to select a profile picture from the gallery, but not much else.

(*) Well, actually, Google doesn't say it will be compulsory. It says that apps "are requested to use _a system picker_ such as the [Android photo picker](https://android-developers.googleblog.com/2023/04/photo-picker-everywhere.html)". I reckon that Essentials uses _a system picker_, although not the [new photo picker](https://developer.android.com/training/data-storage/shared/photopicker), and that that is enough.

To put that to the test, I deselected those permissions from my app Android manifest, made a clean build and made sure that the app was able to pick a photo from the gallery with no issues, with what appears to be a system picker (I did not delve into the Essentials code to make sure of that).

I thought that it would be a nice idea to mention this on this thread to enrich the conversation and to help people looking for information about the implications of this policy change.

**Comment by jfversluis** [comment]
Merged as part of .NET 10 preview 4
